core/src/CompletionStream.ts

```diff
@@ -27,17 +27,33 @@ export class DefaultCompletionStream<PromptT = any> implements CompletionStream<
         const start = Date.now();
         const stream = await this.driver.requestCompletionStream(this.prompt, this.options);
 
+        let finish_reason: string = "";
+        let promptTokens: number = 0;
+        let resultTokens: number = -1;   //-1 Used as a trigger to see if it has been changed
         for await (const chunk of stream) {
             if (chunk) {
-                chunks.push(chunk);
-                yield chunk;
+                if (typeof chunk === 'string'){
+                    chunks.push(chunk);
+                    yield chunk;
+                }else{
+                    if(!(finish_reason == "stop" && !chunk.result)){    //Used to skip empty chunk after stop
```

Comment:
rewrite this if condition. Is to hard to understand - use if else branches if needed.

Reply:
Condition removed and replaced with a simple
 if(chunk.finish_reason){finish_reason = chunk.finish_reason}
Reason for this change is to simplify and because some models/providers use a chunk after the stop chunk for useful information like token count, so cannot skip entirely.